src/query/block/meta.go

```diff
@@ -125,8 +142,9 @@ type ResultMetadata struct {
 	MetadataByName map[string]*ResultMetricMetadata
 }
 
-// ByName returns the ResultMetricMetadata for a given metric name
-func (m ResultMetadata) ByName(name string) *ResultMetricMetadata {
+// ByName returns the ResultMetricMetadata for a given metric name.
+func (m ResultMetadata) ByName(nameTag []byte) *ResultMetricMetadata {
+	name := string(nameTag)
```

Comment:
nit: there's an optimization for keys here to avoid allocs if we use
```golang
r, ok := m.MetadataByName[string(nameTag)]
// ...
m.MetadataByName[string(nameTag)] = r
```
instead of converting to a string, which would alloc.
Example here: https://gist.github.com/stokito/d3a8e26f50c7839b82bd36a4df356e0e

Reply:
Oh, awesome!